### PR TITLE
Handle alternateUrl

### PR DESCRIPTION
**Pull request by zhouyx**
cc/ @avimehta @cramforce

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -77,41 +77,63 @@ function getScopedCid(scope, apiKey, callback) {
...
  */
-function fetchCid(url, scope, token, callback) {
+function fetchCid(apiKey, scope, token, callback) {
   var payload = {'originScope': scope};
   if (token) {
     payload['securityToken'] = token;
   }
+  const url = GOOGLE_API_URL + apiKey;
   fetchJson(url, payload, TIMEOUT, function (err, res) {
     if (err) {
```
Comment (avimehta) [inline index.js]: why not remove `if (err)` also?

File: index.js [hunk 2]
```diff
@@ -77,41 +77,63 @@ function getScopedCid(scope, apiKey, callback) {
...
   });
 }
 
 /**
  * Fetches CID remotely
  *
- * @param {string} url
+ * @param {string} apiKey
  * @param {string} scope
  * @param {string=} token
```
Comment (erwinmombay) [inline index.js]: can you change this to string|undefined

Comment (erwinmombay) [reply index.js]: actually in this case it should `string|null` since `getCookie` explicitly returns `null`

File: index.js (hunk 1 above)
Comment (zhouyx) [reply index.js]: Was trying to check `error` first thing first. But I agree should remove `if (err) ` as well. done

File: index.js (hunk 2 above)
Comment (zhouyx) [reply index.js]: changed to `{?string}`

File: test.js [hunk 3]
```diff
@@ -88,6 +88,41 @@ describe('Test cid.js', function() {
     });
   });
 
+    it('Should respect API responding alternateUrl', function(done) {
```
Comment (avimehta) [inline test.js]: Also a test without token?

**Review by avimehta — APPROVED** [review]

File: test.js (hunk 3 above)
Comment (zhouyx) [reply test.js]: Sure, always good to have more test coverage : )

Comment (zhouyx) [reply test.js]: actually I think that's a duplicate test. test on AMP_TOKEN is already there, we only want to add test to `alternateUrl`. But I added the test anyway : )

**Comment by erwinmombay** [comment]
@zhouyx is this safe to merge?

**Comment by zhouyx** [comment]
Yes, please merge : )

**Pull request merged**